test/ui/selenium/LoginTests.js

```diff
@@ -0,0 +1,252 @@
+const { Assertion } = require('chai');
+const { assert } = require('chai');
+const {Builder, By, TouchSequence} = require('selenium-webdriver');
+const HomePage = require('../POM/HomePage');
+const LoginPage = require('../POM/LoginPage');
+const readline = require('readline');
+const { TsJestCompiler } = require('ts-jest');
+const email = process.env.LOGIN_EMAIL;
+const password = process.env.LOGIN_PASSWORD;
+const adminEmail = process.env.LOGIN_EMAIL_ADMIN;
+const adminPassword = process.env.LOGIN_PASSWORD_ADMIN;
+require('dotenv').config();
+
+const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+});
+
+
+describe('Login Tests', function() {
+
+    let driver;
+    
+    before(async function() {
+        driver = await new Builder().forBrowser('chrome').build();
+    });
+
+    after(async function() {
+        if (driver){
+            await driver.quit();
+        }
+    });
+
+    it('@login should login with vaild login details', async function () {
+
+        this.timeout(100000);
+
+        try {
+            
+            console.log("Test - Valid Login")
+
+            const homePage = new HomePage(driver);
+            const loginPage = new LoginPage(driver);
+
+            await homePage.mainPage();
+            await sleep(2000);
+            console.log(await driver.getTitle());
+
+            await homePage.openLoginPage();
+            await sleep(2000);
+            console.log(await driver.getTitle());
+
+            await loginPage.enterEmail(email);
+            await loginPage.enterPassword(password);
+            await loginPage.clickLoginButton();
+            await sleep(3000);
+            console.log(await driver.getTitle());
+
+            let logout_link = await homePage.logoutLink();
+            let getLogout = await logout_link.getText();
+            assert(getLogout == "Log Out");
+
+        } catch (error) {
+            console.log("error during login: ", error);
+            throw error;
+        } 
+    });
+
+    it('@login should fail login with invalid password', async function () {
+        
+        this.timeout(100000);
+
+        try {
+
+            console.log('Test - Invalid Password');
+
+            const homePage = new HomePage(driver);
+            const loginPage = new LoginPage(driver);
+
+
+            await homePage.mainPage();
+            await sleep(2000);
+            console.log(await driver.getTitle());
+
+            await homePage.openLoginPage();
+            await sleep(2000);
+            console.log(await driver.getTitle());
+
+            await loginPage.enterEmail(email);
+            await loginPage.enterPassword('test');
+            await loginPage.clickLoginButton();
+            await sleep(3000);
+            console.log(await driver.getTitle());
+
+            let elm = await driver.findElement(By.xpath("/html/body/div/div/h3"))
```

Comment:
The element should have an id in the html and you should then be able to get it by id. xpath should be avoided when possible

Reply:
will add ID